src/python/pants/engine/legacy/graph_test.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from textwrap import dedent
+
+from pants.build_graph.address import Address
+from pants.engine.addressable import Addresses
+from pants.engine.rules import RootRule
+from pants.engine.target import (
+    Dependencies,
+    Target,
+    Targets,
+    TransitiveTarget,
+    TransitiveTargets,
+    WrappedTarget,
+)
+from pants.testutil.test_base import TestBase
+from pants.util.ordered_set import FrozenOrderedSet
+
+
+class MockTarget(Target):
+    alias = "target"
+    core_fields = (Dependencies,)
+
+
+class GraphTest(TestBase):
+    @classmethod
+    def rules(cls):
+        return (*super().rules(), RootRule(Addresses), RootRule(WrappedTarget))
+
+    @classmethod
+    def target_types(cls):
+        return (MockTarget,)
+
+    def test_transitive_targets(self) -> None:
+        t1 = MockTarget({}, address=Address.parse(":t1"))
```

Comment:
Adding a test that BUILD file declaration order and graph walk order are respected seems like a good idea since resolvers / users of resolvers rely on this in practice.

Reply:
Aha, missed that - I'll comment there on how the order looks wrong.